source/fuzz/fuzzer_pass_permute_function_variables.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2021 Mostafa Ashraf
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "source/fuzz/fuzzer_pass_permute_function_variables.h"
+#include <numeric>
+#include<vector>
+#include<algorithm>
+
+#include "source/fuzz/fuzzer_context.h"
+#include "source/fuzz/fuzzer_util.h"
+#include "source/fuzz/instruction_descriptor.h"
+
+namespace spvtools {
+namespace fuzz {
+
+FuzzerPassPermuteFunctionVariables::FuzzerPassPermuteFunctionVariables(
+    opt::IRContext* ir_context, TransformationContext* transformation_context,
+    FuzzerContext* fuzzer_context,
+    protobufs::TransformationSequence* transformations)
+    : FuzzerPass(ir_context, transformation_context, fuzzer_context,
+                 transformations) {} // Here we call parent constructor
+
+
+void FuzzerPassPermuteFunctionVariables::Apply() {
+  // here I want to loop over all functions and for each one,
+  // Then deterimne number of OpVariable and swap them.
+for (auto& Func : *GetIRContext()->module()) {
```

Comment:
Variable names should be `snake_cased`.

Reply:
This applies to all variable names. You can check the style guidelines here https://google.github.io/styleguide/cppguide.html.